sdk/keyvault/Azure.Security.KeyVault.Administration/src/BackupOperation.cs

```diff
@@ -116,26 +118,41 @@ public override BackupResult Value
         public override Response GetRawResponse() => _response;
 
         /// <inheritdoc/>
-        public override Response UpdateStatus(CancellationToken cancellationToken = default)
-        {
-            if (!HasCompleted)
-            {
-                Response<FullBackupDetailsInternal> response = _client.GetBackupDetails(Id, cancellationToken);
-                _value = response.Value;
-                _response = response.GetRawResponse();
-            }
-
-            return GetRawResponse();
-        }
+        public override Response UpdateStatus(CancellationToken cancellationToken = default) =>
+            UpdateStatusAsync(false, cancellationToken).EnsureCompleted();
 
         /// <inheritdoc/>
-        public override async ValueTask<Response> UpdateStatusAsync(CancellationToken cancellationToken = default)
+        public override async ValueTask<Response> UpdateStatusAsync(CancellationToken cancellationToken = default) =>
+            await UpdateStatusAsync(true, cancellationToken).ConfigureAwait(false);
+
+        private async ValueTask<Response> UpdateStatusAsync(bool async, CancellationToken cancellationToken = default)
         {
             if (!HasCompleted)
             {
-                Response<FullBackupDetailsInternal> response = await _client.GetBackupDetailsAsync(Id, cancellationToken).ConfigureAwait(false);
-                _value = response.Value;
-                _response = response.GetRawResponse();
+                try
+                {
+                    Response<FullBackupDetailsInternal> response = async ?
+                        await _client.GetBackupDetailsAsync(Id, cancellationToken).ConfigureAwait(false)
+                        : _client.GetBackupDetails(Id, cancellationToken);
+
+                    _value = response.Value;
+                    _response = response.GetRawResponse();
+                }
+                catch (RequestFailedException ex)
+                {
+                    _requestFailedException = ex;
+                    throw;
+                }
+                catch (Exception ex)
+                {
+                    _requestFailedException = new RequestFailedException("Unexpected Failure.", ex);
+                    throw _requestFailedException;
+                }
+                if (_value != null && _value.EndTime.HasValue && _value.Error != null)
+                {
+                    _requestFailedException = new RequestFailedException($"{_value.Error.Message}\nInnerError: {_value.Error.InnerError}\nCode: {_value.Error.Code}");
```

Comment:
Do we not already have a utility method in Core, perhaps, that would do this consistently?

Reply:
Which part were you thinking should be consistent? The error payload in this case is specific to the service.

Reply:
The formatting itself.